core/src/agent/chain_store.rs

```diff
@@ -29,6 +29,19 @@ impl PartialEq for ChainStore {
     }
 }
 
+#[derive(Default, Debug, Clone)]
+pub struct ChainStoreQueryOptions {
+    pub start: usize,
+    pub limit: usize,
+    pub headers: bool,
+}
+
+#[derive(Debug)]
+pub enum ChainStoreQueryResult {
```

Comment:
I think I agree with @maackle that this shouldn't be an enum, but instead a struct with Option<Vec<Address>> and Option<Vec<Chanheader>> that have Some() or None() based on the query so that you don't have to do two queries if you want both.  Then in the request we would have an enum for "Entries, Headers, Both".

It would be fine to have helper functions in the hdk that assume you only want one or the other and do the ChainStoreQueryOptions setting correctly and unrwap to just one or the other..  But the core function should allow both to be returned.

Reply:
@maackle and I just paired; We'll be able to Addresses, Headers, and Entries without too much trouble.  Should have it implemented for review by end of day, hopefully.